core/commonMain/src/kotlinx/io/buffer/BufferOperations.kt

```diff
@@ -66,14 +66,21 @@ inline fun Buffer.copyTo(destination: ByteArray, offset: Long, length: Int, dest
  * to the [destination] at [destinationOffset].
  * Copying bytes from a buffer to itself is allowed.
  */
-fun Buffer.copyTo(destination: Buffer, offset: Long, length: Long, destinationOffset: Long) =
+public fun Buffer.copyTo(destination: Buffer, offset: Long, length: Long, destinationOffset: Long): Unit =
     copyTo(destination, offset.toIntOrFail { "offset" }, length.toIntOrFail { "length" }, destinationOffset.toIntOrFail { "destinationOffset" })
 
+public fun Buffer.toByteArray(offset: Int, length: Int): ByteArray {
+    val result = ByteArray(length)
+    loadByteArray(offset, result, 0, length)
+    return result
+}
+
 /**
  * Execute [block] of code providing a temporary instance of [Buffer] view of this byte array range
  * starting at the specified [offset] and having the specified bytes [length].
  * By default, if neither [offset] nor [length] specified, the whole array is used.
  * An instance of [Buffer] provided into the [block] should be never captured and used outside of lambda.
  */
-expect fun <R> ByteArray.useBuffer(offset: Int = 0, length: Int = size - offset, block: (Buffer) -> R): R
+public expect fun <R> ByteArray.useBuffer(offset: Int = 0, length: Int = size - offset, block: (Buffer) -> R): R
 
+internal expect fun Buffer.compact(startIndex: Int, endIndex: Int): Int
```

Comment:
This function is both unused and untested

Reply:
tnx. Will be moved to first usage place pr